docs/js-core-2/cowsay-project/readme.md

```diff
@@ -0,0 +1,120 @@
+---
+id: readme
+title: Cowsay
+sidebar_label: Cowsay Project
+---
+
+## Cowsay
+
+This coursework assumes you have [Node](https://nodejs.org/en/download/), a text editor, and a terminal or command line.
+
+Cowsay is a fun script that prints an [ASCII](https://simple.wikipedia.org/wiki/ASCII) cow with your words in a speech bubble. It works like this: there is a library of pictures of cows made out of ASCII characters. You call the script in your terminal and you pass in arguments. There are lots of options, but one argument is required: a text string for the cow to say.
+
+```
+ ________
+< Mooooo >
+ --------
+        \   ^__^
+         \  (oo)\_______
+            (__)\       )\/\
+                ||----w |
+                ||     ||
+
+```
+
+Cowsay was originally written in [Perl](https://simple.wikipedia.org/wiki/Perl). In this coursework, we will run it with Node. How would we go about this? What things would be helpful? I think we could use:
+
+* A node package that contains the cowsay library. I'll start you off by letting you know there is one, and it is called ```cowsay```
+* A way to execute the script.
+
+Before we start, what's the fastest way you could get the cow to talk? Well, you can execute some packages directly [in the command line with npx](https://nodejs.dev/learn/the-npx-nodejs-package-runner).
+
+```
+npm install cowsay
+npx cowsay Moooo
+```
+
+We did it! But let's explore some more. How would you install cowsay as a dependency? Make a new folder called /cowsaying and initialise a new package.
+```
+mkdir /cowsaying
```

Comment:
One thing this highlights is that I don't think we have a consistent way we get our trainees to organise code on their laptops, which occasionally causes problems. Maybe we could introduce something like "All of your projects/git repos should be in ~/code" or similar early in the course, to help with this?

We definitely see problems with people accidentally running `npm init` in their homedir and it breaking stuff - having clear guidance here would help to ensure people don't get tripped up...

For now in this exercise, though, maybe ~/cowsaying would be handier than /cowsaying - I suspect the latter may have some permissions issues for some trainees :)